AppInspector.CLI/ResultsWriter.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.ApplicationInspector.Commands;
+using System;
+using System.IO;
+
+namespace Microsoft.ApplicationInspector.CLI
+{
+    /// <summary>
+    /// Wrapper for CLI only output arg validation which may be unique to a command 
+    /// and for allocating the correct writter type and format writter object 
+    /// </summary>
+    public static class ResultsWriter
+    {
+        public static void Write(TagDiffResult tagDiffResult, CLITagDiffCmdOptions options)
```

Comment:
Looking at the writeresults functions it doesnt look like you need all these different wrappers and you could use a single wrapper with the parent classes as the arguments.

Reply:
fixed